app/scripts/directives/oscPersistentVolumeClaim.js

```diff
@@ -26,7 +27,26 @@ angular.module("openshiftConsole")
         }];
         scope.claim.selectedLabels = [];
         DataService.list({group: 'storage.k8s.io', resource: 'storageclasses'}, {}, function(storageClasses) {
-          scope.storageClasses = storageClasses.by('metadata.name');
+          scope.storageClasses = _.map(storageClasses.by('metadata.name'));
+          var defaultClassAnnotation = "storageclass.beta.kubernetes.io/is-default-class";
+          _.each(scope.storageClasses, function(sclass){
+            if ("annotations" in sclass.metadata && defaultClassAnnotation in sclass.metadata.annotations && sclass.metadata.annotations[defaultClassAnnotation] === "true") {
+              scope.defaultStorageClass = sclass.metadata.name;
+            }
+          });
+          if (scope.defaultStorageClass === "")  { //if there is no default, set a no storage class option
```

Comment:
i havent been following the whole thread, but if we are able to know there is a default then 'no storage class' doesn't really make sense as the option.  If we know which storage class is the default I would rather see us pre-select that option and have an indicator like (Default) next to that name.

If we know there is a default but don't know which one is the default, then I would expect "No Storage Class" just becomes "Default storage class"

If we have no way at all to know if there is a default, then yeah we are stuck with "No storage class" and explaining it in the help text that it may select the default if one exists.